models/TestCategoryPresetMap.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTestPreviewer\models;
+
+use oat\oatbox\log\LoggerAwareTrait;
+use oat\taoQtiTest\models\TestCategoryPreset;
+use oat\taoQtiTest\models\TestCategoryPresetProvider;
+use RuntimeException;
+
+class TestCategoryPresetMap
+{
+    use LoggerAwareTrait;
+
+    /** @var TestCategoryPresetProvider */
+    private $testCategoryPresetProvider;
+
+    public function __construct(TestCategoryPresetProvider $testCategoryPresetProvider)
+    {
+        $this->testCategoryPresetProvider = $testCategoryPresetProvider;
+    }
+
+    public function getMap(): array
+    {
+        $presetList = $this->testCategoryPresetProvider
```

Comment:
What happens if the find fails here?  Do we need to catch the exception?

Reply:
It will throw exception, we will catch it on controller level

Reply:
I see. In this case, Do we really need to stop the entire flow because of it? Is it possible to catch it here by logging the error and doing a fallback so the application doesn't crash?

Reply:
I think failure in this case would signal critical issue that will impact instance. I could send empty array but FE will not be able to handle with some of their plugins. @bziondik?

Reply:
I think it is critical issue